UI components

alexandercyu opened the pull request:
Fixes Issue  #2 
Fixes Issue #3 

Fixes issue with dropdowns (list types and panel types) where the chevron rotating and background colour changing is not consistent with the expected function of a dropdown.

alexandercyu committed:
Update stac-ui.css

- Add styles to swap dropdown text colour, background colour, rotate chevron icons when dropdown is expanded or collapsed based on aria-expanded attribute

- remove old styles for swapping text and background colour, and rotating chevron icons

- remove old IDs for Share and Source dropdown panels when checkboxes were used
- NOTE: keep commented styles until they are confirmed not in use

alexandercyu committed:
Update dropdown-panel.html

- Remove javascript used for running functions to rotate chevron icons and swap background colour
- add "title_swap_class" for Source and Share dropdowns because of the STAC version text being swapped in

alexandercyu committed:
Update dropdown.html

- remove javascript used for rotating chevron icon and swapping background colour, text colour
- add classes used to swap text colour for Frequency dropdown

alexandercyu committed:
Update dropdown-default-scripts.js

- remove unneeded functions

alexandercyu committed:
Update dropdown-list.js

Remove empty space

alexandercyu committed:
Delete dropdown-regular-scripts.js

Remove file

alexandercyu committed:
Update ui-styles-scripts.js

Remove unneeded functions

mishaschwartz reviewed (changes requested):
- chevron does not rotate on medium and small dropdowns
- expanded dropdown does not change colour on hover

alexandercyu commented:
> * chevron does not rotate on medium and small dropdowns
> 
>     * expanded dropdown does not change colour on hover

I don't think the chevron on medium and small dropdowns is supposed to rotate.  The design shows it facing the same direction in collapsed and expanded modes.

Small and medium dropdowns only have the list items change colour on hover.  The only one not working is the first item in the small dropdown, which I will fix.

mishaschwartz commented:
> I don't think the chevron on medium and small dropdowns is supposed to rotate. The design shows it facing the same direction in collapsed and expanded modes.

Yup you're right. My mistake.

> Small and medium dropdowns only have the list items change colour on hover. The only one not working is the first item in the small dropdown, which I will fix.

Please look at the design for the first item in the large dropdown when it is expanded

alexandercyu committed:
Update dropdown-default-scripts.js

Add function for a checkbox to check all checkboxes in the dropdown list and display how many are checked in the title of the dropdown.

alexandercyu committed:
Update dropdown.html

- Put the jinja if-else statement back to what it was before. (Just swaps the beginning of the anchor tag)
- Add checkbox at top of list to check all other checkboxes in list

alexandercyu committed:
Update stac-ui.css

- Fixed colour change on hover for metadata dropdown
- Fixed hover background colour for small dropdown
- Fixed hover text colour for small and medium dropdown
- Remove commented code
- Change border radius style for medium dropdown list to use pure CSS instead of javascript function

alexandercyu committed:
Update dropdown.html

Remove swapBorderRadius function

alexandercyu committed:
Update dropdown-list.js

Remove swapBorderRadius function

alexandercyu commented:
> > I don't think the chevron on medium and small dropdowns is supposed to rotate. The design shows it facing the same direction in collapsed and expanded modes.
> 
> Yup you're right. My mistake.
> 
> > Small and medium dropdowns only have the list items change colour on hover. The only one not working is the first item in the small dropdown, which I will fix.
> 
> Please look at the design for the first item in the large dropdown when it is expanded

Made the metadata dropdown's first item change colour on hover.

Fixed small dropdown's first item so it changes colour on hover.

alexandercyu committed:
Update stac-ui.css

Fix style for medium dropdown

alexandercyu committed:
Update dropdown.html

Move script to inside the anchor tag so the CSS last-child will work on the list

alexandercyu committed:
Update stac-ui.css

Removed unused css

alexandercyu committed:
Update dropdown-panel.html

Removed unused classes

alexandercyu committed:
Update sample.html

Added space between dropdown panel samples

alexandercyu committed:
Update stac-ui.css

Remove spaces

alexandercyu committed:
Update stac-ui.css

- Fixed border radius  in default dropdown
- fixed hover colour and font colour in Frequency dropdown

alexandercyu commented:
Additionally fixed: 
- the default dropdown's inner border radius
- the Frequency dropdown's text colour change on expand
- removed unneeded CSS

mishaschwartz reviewed (changes requested):
Please implement THIS:

![image](https://github.com/user-attachments/assets/8be30252-46f4-4915-8d54-ae286654bf36)
